### PR TITLE
Add security tests for cross clusters query

### PR DESCRIPTION
**Pull request by dnhatn**
This is a follow-up to #104840, where security was deferred so that it can be reviewed separately. The action `cluster:monitor/xpack/enrich/esql/resolve_policy` can now be called across clusters to resolve enrich policies on remote clusters. This PR  registers the action for reading from remote clusters and adds a security test for that.

Relates #104840

### CODE REVIEW

**Comment by dnhatn** [comment]
@jakelandis Could you please take a look? Thank you!

**Review by jakelandis — COMMENTED** [review]
Changes look good. Can you also add a test for RCS 2.0 (the security model that requires an API key) ?

It should be pretty straight forward as a Java based test here: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java . (or clone it and create a standalone ESQL/CCS specific test). You shouldn't need to change the security setup much since `read_cross_cluster` is [automatically](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/CrossClusterApiKeyRoleDescriptorBuilder.java#L29) added to the API key used between clusters. You will need to add the enrich module to the (fulfilling) test setup and either reuse an existing index or add a new index to remote_indices on the querying cluster.

**Comment by dnhatn** [comment]
Thanks Jake! I added a test for RCS 2.0, but it seems the `monitor_enrich` cluster privilege doesn't propagate to the remote cluster. I think I am missing something here. Can you please take a look at the failure?

```
 ./gradlew ':x-pack:plugin:security:qa:multi-cluster:javaRestTest' --tests "org.elasticsearch.xpack.remotecluster.RemoteClusterSecurityRestIT.testCrossClusterEnrich" -Dtests.seed=D49C98BB5556C7B3 -Dtests.locale=und -Dtests.timezone=America/Rankin_Inlet -Druntime.java=21
```

```
org.elasticsearch.xpack.remotecluster.RemoteClusterSecurityRestIT > testCrossClusterEnrich FAILED
    org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:60262], URI [_query], status line [HTTP/1.1 403 Forbidden]
    {"error":{"root_cause":[{"type":"security_exception","reason":"action [cluster:monitor/xpack/enrich/esql/resolve_policy] towards remote cluster is unauthorized for user [remote_search_user] with assigned roles [remote_search] authenticated by API key id [TsX3W40BtyOttFJQ4Lnw] of user [test_user], this action is granted by the cluster privileges [monitor_enrich,manage_enrich,monitor,manage,all]"}],"type":"security_exception","reason":"action [cluster:monitor/xpack/enrich/esql/resolve_policy] towards remote cluster is unauthorized for user [remote_search_user] with assigned roles [remote_search] authenticated by API key id [TsX3W40BtyOttFJQ4Lnw] of user [test_user], this action is granted by the cluster privileges [monitor_enrich,manage_enrich,monitor,manage,all]"},"status":403}
        at __randomizedtesting.SeedInfo.seed([D49C98BB5556C7B3:F2AB3F3E7D177935]:0)
        at app//org.elasticsearch.client.RestClient.convertResponse(RestClient.java:351)
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:317)
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:292)
        at app//org.elasticsearch.xpack.remotecluster.RemoteClusterSecurityRestIT.performRequestWithRemoteSearchUser(RemoteClusterSecurityRestIT.java:688)
        at app//org.elasticsearch.xpack.remotecluster.RemoteClusterSecurityRestIT.testCrossClusterEnrich(RemoteClusterSecurityRestIT.java:580)
```

**Comment by dnhatn** [comment]
@jakelandis I will give it another try today.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java [hunk 1]
```diff
@@ -82,7 +82,8 @@ public final class IndexPrivilege extends Privilege {
         // cross clusters query for ESQL
         "internal:data/read/esql/open_exchange",
         "internal:data/read/esql/exchange",
-        "indices:data/read/esql/cluster"
+        "indices:data/read/esql/cluster",
+        "cluster:monitor/xpack/enrich/esql/resolve_policy"
```
Comment (jakelandis) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java:86]: this should be part of a named cross cluster *cluster* privilege (not the index privilege) : https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilegeResolver.java#L166

Comment (dnhatn) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java]: Yes, fixed in https://github.com/elastic/elasticsearch/pull/104879/commits/7cfeb7242e720871a87c571b02917096b3e53cc9.

**Review by mark-vieira — APPROVED** [review]

**Comment by dnhatn** [comment]
@jakelandis @n1v0lg I merging this PR with two RCS2.0 tests muted. Thank you for the feedback.

@mark-vieira Thanks for review.

**Pull request merged**